src/hotspot/share/oops/constantPool.cpp

```diff
@@ -388,44 +373,69 @@ void ConstantPool::remove_unshareable_info() {
     resolved_references() != NULL ? resolved_references()->length() : 0);
   set_resolved_references(OopHandle());
 
-  int num_klasses = 0;
+  bool archived = false;
```

Comment:
I think this declaration could be moved to line 392 since it is only used in that case.

Reply:
Fixed